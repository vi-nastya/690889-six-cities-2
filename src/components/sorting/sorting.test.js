import React from "react";
import renderer from "react-test-renderer";
import Sorting, {SORT_TYPES} from "./sorting";

it(`Sorting component is rendered correctly after relaunch`, () => {
  const tree = renderer.create(<Sorting isOpen={true}
    activeSortType={SORT_TYPES[0]}
    toggleSortListHandler={jest.fn()}
    sortTypeClickHandler={jest.fn()}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
