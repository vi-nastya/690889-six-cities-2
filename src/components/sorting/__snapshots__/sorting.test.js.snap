// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting component is rendered correctly after relaunch 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
    onClick={[MockFunction]}
  >
    <li
      className="places__option "
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
